runtime/lua/vim/treesitter/dev.lua

```diff
@@ -436,11 +457,153 @@ function M.inspect_tree(opts)
     buffer = buf,
     once = true,
     callback = function()
-      if api.nvim_win_is_valid(w) then
-        api.nvim_win_close(w, true)
+      close_win(w)
+    end,
+  })
+end
+
+-- between the query editor and the tree inspector.
+--- @class EditQueryOpts
+--- @field bufnr integer? Buffer to draw the editor into. If omitted, a new
+---                       buffer is created.
+--- @field winid integer? Window id to display the editor buffer in. If omitted,
+---                       a new window is created with {command}.
+--- @field command string? Vimscript command to create the window. Default
+---                        value is "60vnew" or "new" if the inspector is open.
+---                        Only used when {winid} is nil.
+
+--- @private
+--- @param opts EditQueryOpts?
+M.edit_query = function(opts)
+  vim.validate({
+    opts = { opts, 't', true },
+  })
+
+  opts = opts or {}
+
+  local buf = api.nvim_get_current_buf()
+
+  -- Close any existing editor window
+  if vim.b[buf].dev_edit then
+    close_win(vim.b[buf].dev_edit)
+  end
+
+  local w = opts.winid
+  if not w then
+    local cmd = opts.command or '60vnew'
+    -- If the inspector is open, place the editor above it.
+    local base_buf = vim.b[buf].dev_base ---@type integer?
+    local inspect_buf = base_buf and vim.b[base_buf].dev_inspect
+    if base_buf and api.nvim_win_is_valid(inspect_buf) then
+      vim.api.nvim_set_current_win(inspect_buf)
+      buf = base_buf
+      cmd = 'new'
+    end
+    vim.cmd(cmd)
+    w = api.nvim_get_current_win()
+  end
+
+  local b = opts.bufnr
+  if b then
+    api.nvim_win_set_buf(w, b)
+  else
+    b = api.nvim_win_get_buf(w)
+  end
+
+  vim.b[buf].dev_edit = w
+  vim.bo[b].omnifunc = 'v:lua.vim.treesitter.query.omnifunc'
+  set_dev_properties(w, b)
+
+  local ok, parser = pcall(vim.treesitter.get_parser, buf)
+  if not ok then
+    return nil, 'No parser available for the given buffer'
+  end
+  local lang = parser:lang()
+
+  -- HACK: omnifunc guesses the language based on the containing folder,
+  -- so add the parser's language to the buffer's name.
+  api.nvim_buf_set_name(b, string.format('%s/query_editor.scm', lang))
+
+  local ns = api.nvim_create_namespace('')
```

Comment:
Anonymous namespaces don't show up in `nvim_get_namespaces()` and can make debugging/introspection more difficult.

```suggestion
  local ns = api.nvim_create_namespace('treesitter/dev-edit')
```

I did not do this myself when creating the namespace for the tree view, but I wish I had (and might go back and change that).

Reply:
I'll piggy-back the change as part of this PR :)

Reply:
Again, maybe enough to have a common ts-dev namespace?